File: server/app/schemas/media.py
"""Media request/response schemas."""

from __future__ import annotations

import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict

from app.models.enums import MediaType, ProcessingStatus


class MediaResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    survey_id: uuid.UUID
    media_type: MediaType
    processing_status: ProcessingStatus
    room_location: str | None
    width: int | None
    height: int | None
    duration_seconds: Decimal | None
    frame_number: int | None
    parent_video_id: uuid.UUID | None
    upload_timestamp: datetime
    # Short-lived signed URL for reading the object; populated on detail/list,
    # omitted on the raw upload acknowledgement.
    url: str | None = None


class MediaUploadResponse(BaseModel):
    items: list[MediaResponse]
    count: int

File: client/app/src/main/java/com/packingandmoving/surveyagent/model/Media.kt
package com.packingandmoving.surveyagent.model

import kotlinx.serialization.Serializable

/**
 * MediaResponse (openapi.json). `url` is a short-lived signed GCS URL, present only on
 * GET /media/{id} and list responses (not on the upload ack) — assume it can expire
 * (~15 min) and re-fetch when needed (frontend-integration.md §11).
 */
@Serializable
data class Media(
    val id: String,
    val surveyId: String,
    val mediaType: MediaType,
    val processingStatus: ProcessingStatus,
    val roomLocation: String?,
    val width: Int?,
    val height: Int?,
    val durationSeconds: String?,
    val frameNumber: Int?,
    val parentVideoId: String?,
    val uploadTimestamp: String,
    val url: String? = null,
)

/** MediaUploadResponse (POST images/videos ack). */
@Serializable
data class MediaUploadResult(
    val items: List<Media>,
    val count: Int,
)
